code/game/objects/items/defib.dm

```diff
@@ -0,0 +1,704 @@
+
+	//placed on chest and short delay to shock for dramatic effect, revive time is 5sec total
+	if(!do_after(user, chargetime, act_target = H))
+		return
+
+	//deduct charge here, in case the base unit was EMPed or something during the delay time
+	if(!checked_use(chargecost))
+		make_announcement("buzzes, \"Insufficient charge.\"", "warning")
+		playsound(get_turf(src), 'sound/machines/defib_failed.ogg', 50, 0)
+		return
+
+	H.visible_message(SPAN_WARNING("\The [H]'s body convulses a bit."))
+	playsound(get_turf(src), "bodyfall", 50, 1)
+	playsound(get_turf(src), 'sound/machines/defib_zap.ogg', 50, 1, -1)
+	set_cooldown(cooldowntime)
+
+	error = can_revive(H)
+	if(error)
+		make_announcement(error, "warning")
+		playsound(get_turf(src), 'sound/machines/defib_failed.ogg', 50, 0)
+		return
+	H.apply_damage(burn_damage_amt, DAMAGE_BURN, BP_CHEST)
+
+	//set oxyloss so that the patient is just barely in crit, if possible
+	make_announcement("pings, \"Resuscitation successful.\"", "notice")
+	playsound(get_turf(src), 'sound/machines/defib_success.ogg', 50, 0)
+	H.resuscitate()
+	var/obj/item/organ/internal/cell/potato = H.internal_organs_by_name[BP_CELL]
+	if(istype(potato) && potato.cell)
+		var/obj/item/cell/C = potato.cell
+		C.give(chargecost)
+	H.AdjustSleeping(-60)
+	log_and_message_admins("used \a [src] to revive [key_name(H)].")
+
+/obj/item/shockpaddles/proc/lowskill_revive(mob/living/carbon/human/H, mob/living/user)
+	if(prob(60))
+		playsound(get_turf(src), 'sound/machines/defib_zap.ogg', 100, 1, -1)
+		H.electrocute_act(burn_damage_amt*4, src, def_zone = BP_CHEST)
+		user.visible_message(SPAN_WARNING("<i>The paddles were misaligned! \The [user] shocks [H] with \the [src]!</i>"), SPAN_WARNING("The paddles were misaligned! You shock [H] with \the [src]!"))
+		return 0
+	if(prob(50))
+		playsound(get_turf(src), 'sound/machines/defib_zap.ogg', 100, 1, -1)
+		user.electrocute_act(burn_damage_amt*2, src, def_zone = BP_L_HAND)
+		user.electrocute_act(burn_damage_amt*2, src, def_zone = BP_R_HAND)
+		user.visible_message(SPAN_WARNING("<i>\The [user] shocks themselves with \the [src]!</i>"), SPAN_WARNING("You forget to move your hands away and shock yourself with \the [src]!"))
+		return 0
+	return 1
+
+/obj/item/shockpaddles/proc/do_electrocute(mob/living/carbon/human/H, mob/user, target_zone)
+	var/obj/item/organ/external/affecting = H.get_organ(target_zone)
+	if(!affecting)
+		to_chat(user, SPAN_WARNING("They are missing that body part!"))
+		return
+
+	//no need to spend time carefully placing the paddles, we're just trying to shock them
+	user.visible_message(SPAN_DANGER("\The [user] slaps [src] onto [H]'s [affecting.name]."), SPAN_DANGER("You overcharge [src] and slap them onto [H]'s [affecting.name]."))
+
+	//Just stop at awkwardly slapping electrodes on people if the safety is enabled
+	if(safety)
+		to_chat(user, SPAN_WARNING("You can't do that while the safety is enabled."))
+		return
```

Comment:
shouldnt check be performed before displaying a visible message about overcharging the defi?

Reply:
intentional as per the comment